apps/cow-fi/hooks/useLazyLoadImages.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useRef, useCallback } from 'react'
+
+// Shared IntersectionObserver instance
+let observer: IntersectionObserver | null = null
+
+const LAZY_LOADING_CONFIG = {
+  rootMargin: '25px',
+  placeholderColor: '#f0f0f0',
+  fadeInDuration: '0.3s',
+  minHeight: '100px',
+}
+
+// Utility function to generate placeholder src
+const getPlaceholderSrc = (placeholderColor: string): string => {
+  const encodedColor = encodeURIComponent(placeholderColor)
+  return `data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' width='100%' height='100%' viewBox='0 0 1 1' preserveAspectRatio='none'><rect width='1' height='1' fill='${encodedColor}' /></svg>`
+}
+
+// Initialize a shared IntersectionObserver
+const initObserver = (onIntersect: (entry: IntersectionObserverEntry) => void) => {
+  if (observer) return
+
+  observer = new IntersectionObserver(
+    (entries) => {
+      entries.forEach(onIntersect)
+    },
+    { rootMargin: LAZY_LOADING_CONFIG.rootMargin },
+  )
+}
+
+export function useLazyLoadImages() {
+  const placeholderSrc = useRef(getPlaceholderSrc(LAZY_LOADING_CONFIG.placeholderColor)).current
+
+  const replaceImageUrls = useCallback(
+    (html: string): string => {
+      return html.replace(/<img([^>]*)src="([^"]*)"([^>]*)>/g, (_, before, src, after) => {
+        return `<img${before}data-src="${src}" src="${placeholderSrc}"${after}>`
+      })
+    },
+    [placeholderSrc],
+  )
+
+  const LazyImage: React.FC<React.ImgHTMLAttributes<HTMLImageElement>> = ({
+    src,
+    alt = '',
+    width,
+    height,
+    style,
+    ...props
+  }) => {
+    const imgRef = useRef<HTMLImageElement>(null)
+
+    useEffect(() => {
+      if (!imgRef.current || !src) return
+
+      const handleIntersect = (entry: IntersectionObserverEntry) => {
+        const img = entry.target as HTMLImageElement
+        if (entry.isIntersecting && img.dataset.src) {
+          img.src = img.dataset.src
+          img.style.opacity = '1'
+          observer?.unobserve(img)
+        }
+      }
+
+      initObserver(handleIntersect)
```

Comment:
I think `initObserver` function doesn't work right.
So, it creates `IntersectionObserver` only once regardless of a context.
It means that if you call the `useLazyLoadImages()` hook once, in the next time it won't work and `handleIntersect` won't be called